WallProject/WallProject/Controllers/HomeController.cs

```diff
@@ -14,24 +15,27 @@ public class HomeController : Controller
     {
         private readonly ILogger<HomeController> _logger;
         private readonly IWallService _service;
+        private readonly ICommentService _commentService;
+        private readonly IPostService _postService;
 
-        public HomeController(ILogger<HomeController> logger, IWallService service)
+        public HomeController(ILogger<HomeController> logger, IWallService service, ICommentService commentService, IPostService postService)
         {
             _logger = logger;
-           _service = service;
+            _service = service;
+            _commentService = commentService;
+            _postService = postService;
         }
 
         public async Task<IActionResult> WallAsync()
         {
-           
-            PersonViewModel user= await _service.getUser();
-            WallViewModel wall = new WallViewModel();
-            wall.Owner = user;
-            //Bedzie uzupełniane jak beda w bazie
-            wall.Posts = new List<PostViewModel>();
-            return View(wall);
+            ServiceResult<WallViewModel> wall = await _service.getWall(1);
+            if (wall.IsOk())
+                return View(wall.Result);
+            else
+                return View("Privacy");
```

Comment:
Ma nie być logiki w kontrolerze!
Zamiast robienia ifów to ma byc klasa która automatycznie zwraca View na podstawie tego co dostała w service Result
Tak samo jak w web api

Reply:
okej, ale czy wiesz że zwracanie View jest cechą typową dla kontrolerów?
żeby to zadziałało klasa zwracająca View do kontrolera musiałaby dziedziczyć ( : Controller ) co według nie czyni ją kontrolerem a w efekcie dokładamy po prostu jeszcze jeden kontroler po drodze tylko że nie w tym końcowym jest if tylko w tym poprzednim... więc kolejna warstwa...